### PR TITLE
fix(gnovm): post loopvar fixes

### PR DESCRIPTION
**Pull request by thehowl**
These are a couple of fixes after https://github.com/gnolang/gno/pull/5187.

The two commits are separate fixes but grouped together as they are on the same set of commits:

- A first fix tackles the `.loopvar` prefix when present in struct literals. This was incorrectly transcribed; the fix in preprocess.go correctly handles cases like those in `gnovm/tests/files/loopvar_struct_field_*.gno`.
- A second fix is a follow-up from a conversation on the PR; instead of panicking at runtime for const vars with incorrect initializations, we check these at preprocessing.

Finally a set of new tests are added to check more edge cases of loopvar.

### CODE REVIEW

**Comment by Gno2D2** [comment]
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: thehowl/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

File: gnovm/pkg/gnolang/preprocess.go [hunk 1]
```diff
@@ -1241,7 +1241,16 @@ func preprocess1(store Store, ctx BlockNode, n Node) Node {
...
+						// Struct field keys are not variable references.
+						// Strip any .loopvar suffix that replaceAllLoopvar may have
+						// added (it cannot distinguish struct keys from map keys at
+						// the time of the rename pass).
+						fieldName := n.Name
+						if strings.HasSuffix(string(fieldName), ".loopvar") {
+							fieldName = Name(strings.TrimSuffix(string(fieldName), ".loopvar"))
+							n.Name = fieldName
+						}
+						n.Path = bt.GetPathForName(fieldName)
```
Comment (ltzmaxwell) [inline gnovm/pkg/gnolang/preprocess.go]: seems a bit cumbersome to rename and rename back.  
by simply skip composite key in `initStaticBlocks()` seems works fine.

Comment (ltzmaxwell) [reply gnovm/pkg/gnolang/preprocess.go]:
```diff
diff --git a/gnovm/pkg/gnolang/preprocess.go b/gnovm/pkg/gnolang/preprocess.go
index 34f76e4f4..332a73f70 100644
--- a/gnovm/pkg/gnolang/preprocess.go
+++ b/gnovm/pkg/gnolang/preprocess.go
@@ -219,6 +219,8 @@ func initStaticBlocks1(store Store, ctx BlockNode, nn Node) {
                              switch n := n.(type) {
                              case *NameExpr:
                                      switch ftype {
+                                     case TRANS_COMPOSITE_KEY:
+                                             return n, TRANS_CONTINUE
                                        case TRANS_ASSIGN_LHS:
                                                as := ns[len(ns)-1].(*AssignStmt)
                                                if as.Op == DEFINE {
@@ -319,32 +321,6 @@ func initStaticBlocks1(store Store, ctx BlockNode, nn Node) {
```

**Comment by thehowl** [comment]
Closed in favor of #5266
